recipe/reset_from_gitlab_artifact.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Deployer;
+use SourceBroker\DeployerExtendedDatabase\Utility\ConsoleUtility as DatabaseConsoleUtility;
+
+task('reset:from_gitlab_artifact', function () {
+    // set in deploy.php as https://gitlab.example.org/api/v4/projects/<project-id>/jobs/artifacts/<branch>/download?job=<job-of-artifact>
+    $url = get('reset_gitlab_artifact_url');
+    // Gitlab API token for the repository where the artifact is stored. Can be a project token.
+    $gitlabApiToken = (new DatabaseConsoleUtility())->getOption('token', true);
+    // Database dumpcode that was used during the creation of the Gitlab artifact.
+    $dumpCode = (new DatabaseConsoleUtility())->getOption('dumpcode', true);
+
+    if ( filter_var($url, FILTER_VALIDATE_URL) && preg_match('#jobs\/artifacts\/.+\/download$#', parse_url($url, PHP_URL_PATH)) ) {
+        if (get('is_argument_host_the_same_as_local_host')) {
+            $activeDir = get('deploy_path') . (testLocally('[ -e {{deploy_path}}/release ]') ? '/release' : '/current');
+            $activeDir = testLocally('[ -e ' . $activeDir . ' ]') ? $activeDir : get('deploy_path');
+            runLocally('cd ' . $activeDir . ' && curl --location --output artifacts.zip --header "PRIVATE-TOKEN: ' . $gitlabApiToken . '" "' . $url . '"');
+            runLocally('cd ' . $activeDir . ' && vendor/bin/dep db:rmdump {{argument_host}} --options=dumpcode:' . $dumpCode . ' --no-interaction');
+            runLocally('cd ' . $activeDir . ' && unzip -o artifacts.zip');
+            runLocally('cd ' . $activeDir . ' && mv -n .dep/database/dumps/*dumpcode=' . $dumpCode . '* {{db_storage_path_local}}/');
+            runLocally('cd ' . $activeDir . ' && vendor/bin/dep db:decompress {{argument_host}} --options=dumpcode:' . $dumpCode . ' --no-interaction');
+            runLocally('cd ' . $activeDir . ' && vendor/bin/dep db:import {{argument_host}} --options=dumpcode:' . $dumpCode . ' --no-interaction');
+            runLocally('cd ' . $activeDir . ' && vendor/bin/dep db:rmdump {{argument_host}} --options=dumpcode:' . $dumpCode . ' --no-interaction');
+            runLocally('cd ' . $activeDir . ' && rm -f artifacts.zip');
+            runLocally('cd ' . $activeDir . ' && {{local/bin/php}} {{bin/typo3cms}} cache:flush');
+            runLocally('cd ' . $activeDir . ' && {{local/bin/php}} {{bin/typo3cms}} cache:warmup');
+        } else {
+            $verbosity = (new DatabaseConsoleUtility())->getVerbosityAsParameter();
+            run('cd {{release_or_current_path}} && {{bin/php}} {{bin/deployer}} reset:from_gitlab_artifact ' . $verbosity . ' --options="token:' . $gitlabApiToken . ',dumpcode:' . $dumpCode . '" {{argument_host}}');
+        };
+    } else {
+        writeln('Gitlab API URL is invalid: ' . $url);
+        exit(1);
```

Comment:
Better use `error()`: https://deployer.org/docs/7.x/api#error

Reply:
As discussed, no output is provided by this function. However, I will resort to GracefulShutdownException as used by Sourcebroker: https://github.com/sourcebroker/deployer-extended-database/blob/53e3d0fe531d6b7d13d8a55934bb1895101842d7/deployer/db/task/db_push.php#L15